sound/soc/sof/intel/hda-ctrl.c

```diff
@@ -263,3 +263,68 @@ int hda_dsp_ctrl_init_chip(struct snd_sof_dev *sdev, bool full_reset)
 
 	return ret;
 }
+
+void hda_dsp_ctrl_stop_chip(struct snd_sof_dev *sdev)
+{
+	struct hdac_bus *bus = sof_to_bus(sdev);
+	struct hdac_stream *stream;
+	int sd_offset;
+
+	if (!bus->chip_init)
+		return;
+
+	/* disable interrupts in stream descriptor */
+	list_for_each_entry(stream, &bus->stream_list, list) {
+		sd_offset = SOF_STREAM_SD_OFFSET(stream);
+		snd_sof_dsp_update_bits(sdev, HDA_DSP_HDA_BAR,
+					sd_offset +
+					SOF_HDA_ADSP_REG_CL_SD_CTL,
+					SOF_HDA_CL_DMA_SD_INT_MASK,
+					0);
+	}
+
+	/* disable SIE for all streams */
+	snd_sof_dsp_update_bits(sdev, HDA_DSP_HDA_BAR, SOF_HDA_INTCTL,
+				SOF_HDA_INT_ALL_STREAM,	0);
+
+	/* disable hda bus irq */
```

Comment:
disable controller CIE and GIE

Reply:
OK, thanks!

Reply:
no, not disable CIE and GIE,  please check this commit https://github.com/torvalds/linux/commit/7d4f606c50ffaaa3ac60b7faf770dc6e84af3207.
Codec can send out INT or wake up controller depending on whether CIE or GIE enabled.(Figure 4, Interupt structure).

Reply:
@RanderWang Please check 
azx_stop_chip
=>snd_hdac_bus_stop_chip
==>azx_int_disable

Reply:
yes,  I tried this PR with my 983 PR for headphone detection on whiskylake. Headphone detection worked with it.